### PR TITLE
Headers

### PR DESCRIPTION
**Pull request by ZRich97**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+*.DS_Store*
```
Comment (kramsey458) [inline .gitignore:1]: nice items to add to .gitignore

File: headers/app.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+#         arg = list of CSV file paths
+# Returns: The classification metadata
+# TODO: More clearly specify what metadata to return
+def classify(ontology, *arg):
+    # Verify parameters
+    if len(arg) == 0:
+        print_usage()
+    my_classifier = cfr.Classifier(arg)
+    results = my_classifier.classify_ontology(ontology)
+    return results
```
Comment (kramsey458) [inline headers/app.py]: return my_classifier.classify_ontology(ontology)

File: headers/classifier.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    # Params: ontology = JSON object passing 
+    # Returns: classification metadata
+    def classify_ontology(self, ontology):
+        # Once I deconstruct the JSON object (the ontology) I get passed, 
+        # I will use that for comparison. For now just compare based on
+        # similarity between headers. 
+        results = []
+        for filepath in data_filepaths:
+            results.append(hdr.cmp_hdrs(filepath))
+        return []
```
Comment (kramsey458) [inline headers/classifier.py]: return result?

File: headers/headers.py [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+# Params: filepath = path to given CSV file as a String
+#         input_str = given String to which we compare
+# Return: [(other_header, value), ... ]
+def cmp_str_to_hdr(input_str, filepath):
+    headers = get_headers(filepath)
+    total_dists = []
+    for header in headers:
+        lev_dist = calc_lev_dist(input_str, header)
+        ham_dist = calc_ham_dist(input_str, header)
```
Comment (brobinson15) [inline headers/headers.py]: Optimizing the distance between you and any potential ham is a MAJOR key

**Review by kramsey458 — CHANGES_REQUESTED** [review]
Looks good to merge, just minor changes

File: headers/tests.py [hunk 5]
```diff
@@ -0,0 +1,12 @@
+import unittest
+import sys
+from headers import *
+
+class TestBoardStupid(unittest.TestCase):
+
+    def test_get_headers_FNFE(self):
+        self.assertRaises(FileNotFoundError, get_headers, "file_lol.lol")
```
Comment (brobinson15) [inline headers/tests.py]: lol

**Review by kramsey458 — CHANGES_REQUESTED** [review]
Nice code

File: headers/app.py (hunk 2 above)
Comment (tonyc856) [reply headers/app.py]: I think it's fine the way it is. It's for code readability.

File: headers/classifier.py (hunk 3 above)
Comment (ZRich97) [reply headers/classifier.py]: tru

**Comment by tonyc856** [comment]
Nice work on providing the comments.

**Review by brobinson15 — APPROVED** [review]
Much approved.

**Review by kramsey458 — APPROVED** [review]

**Review by brobinson15 — APPROVED** [review]

**Review by brobinson15 — APPROVED** [review]

**Review by brobinson15 — APPROVED** [review]

**Review by brobinson15 — APPROVED** [review]

**Comment by brobinson15** [comment]
Closing due to unsigned commits (see #4)

**Pull request closed without merging**